jerry-core/api/jerry.c

```diff
@@ -465,16 +470,55 @@ jerry_parse (const jerry_char_t *resource_name_p, /**< resource name (usually a
 
   if (JERRY_UNLIKELY (bytecode_data_p == NULL))
   {
+#if ENABLED (JERRY_MODULE_SYSTEM)
+    if ((parse_opts & JERRY_PARSE_MODULE) != 0)
+    {
+      ecma_module_cleanup_context ();
+    }
+#endif /* ENABLED (JERRY_MODULE_SYSTEM) */
+
     return ecma_create_error_reference_from_context ();
   }
 
+#if ENABLED (JERRY_MODULE_SYSTEM)
+  if ((parse_opts & JERRY_PARSE_MODULE) != 0)
+  {
+    ecma_module_t *root_module_p = JERRY_CONTEXT (module_current_p);
```

Comment:
If there is an error below: shouldn't we do a cleanup on the `root_module_p`? or the cleanup_context below takes care of this?